### DISCUSSION TITLE
How do I specify the ssh identity key?

### DISCUSSION THREAD

**Discussion by ychaouche** [axkibe/lsyncd#712] [Problems]
Hello there,
I'm using an old version of lsyncd
(2.1.5)
and I would like to know how do I specify the ssh identity file ?
it seems to be trying only the rsa version,
but since I updated the remote host,
the newer version of SSH doesn't support rsa anymore,
and I had to create another key.
lsyncd didn't pick it up automatically,
and ssh.identityFile isn't recognized 

```

  ssh = {
     port  = "22",
     identityFile = "/home/ychaouche/.ssh/id_ed25519._pub"
   },
```

```
 $ sudo lsyncd -nodaemon /etc/lsyncd.conf.one
Error: error preparing /etc/lsyncd.conf.one: Parameter "ssh.identityFile" unknown. (if this is not a typo add it to checkgauge)
$ 
```

**Comment by juangacovas** [comment]
Tried with _extra into "ssh" section of lsyncd conf? See docs

**Comment by ferdnyc** [comment]
The problem may also be that, when run under `sudo`, `ssh` likely won't accept identity files from `/home/someuser/.ssh/`. Can I ask why `lsyncd` is being run as `root`, when the identity being used is some user's credentials? Why not run `lsyncd` as that user?

Oh, and also, in case it was missed, `ssh` should be under the `sync` configuration object. This definitely works, on my system, when running `lsyncd` as `ferd`:

```js

sync {
    default.rsyncssh,
    ssh = {
        identityFile="/home/ferd/.ssh/lsyncd-id_rsa",
    },
    /* more configs... */
}
```

**Comment by ferdnyc** [comment]
If `lsyncd` _must_ be run as root, it's probably better to store the key in `/root/.ssh/`, with correct permissions (`600` or `660`, owned by `root:root`), and load it from there.
